pymc/logprob/order.py

```diff
@@ -126,3 +138,23 @@ def max_logprob(op, values, base_rv, **kwargs):
     logprob = (n - 1) * logcdf + logprob + pt.math.log(n)
 
     return logprob
+
+
+@_logprob.register(MeasurableMaxDiscrete)
+def max_logprob_discrete(op, values, base_rv, **kwargs):
+    r"""Compute the log-likelihood graph for the `Max` operation.
+
+    The formula that we use here is :
+        \ln(f_{(n)}(x)) = \ln(F(x)^n - F(x-1)^n)
+    where f(x) represents the p.d.f and F(x) represents the c.d.f of the distrivution respectively.
+    """
+    (value,) = values
+    logprob = _logprob_helper(base_rv, value)
+    logcdf = _logcdf_helper(base_rv, value)
+    logcdf_prev = _logcdf_helper(base_rv, value - 1)
+
+    n = base_rv.size
+
+    logprob = pt.log((pt.exp(logcdf)) ** n - (pt.exp(logcdf_prev)) ** n)
```

Comment:
```suggestion
    # TODO: Test if it works when x = 0, for non-negative dist like Poisson
    logprob = pt.logdiffexp(n * logcdf, n * logcdf_prev)
```

Reply:
Note: `logdiffexp` is from the [`pymc.math` submodule](https://github.com/pymc-devs/pymc/blob/main/pymc/math.py#L261-L263)